test/boot_storage_test.rb

```diff
@@ -89,6 +89,18 @@
 
       expect(subject.mbr_disk).to eq "/dev/vda"
     end
+
+    it "skips cache if storage gets changed" do
+      subject.RootPartitionDevice = "/dev/sda1"
+      subject.BootPartitionDevice = "/dev/sda2"
+      subject.mbr_disk = "/dev/sda"
+
+      allow(Yast::Storage).to receive(:GetTargetChangeTime).and_return(Time.now.to_i)
+
+      subject.detect_disks
+
+      expect(subject.RootPartitionDevice).to eq "/dev/vda1"
+    end
```

Comment:
Now that I saw the previous test passing even it was wrong, I'm kind of worried. Can we have a complementary test or something?